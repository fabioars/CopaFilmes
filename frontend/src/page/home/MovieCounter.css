.MovieCounter {
    display: flex;
    flex-direction: column;
    font-size: 26px;
    color: #fff;
    font-weight: bold;
}
