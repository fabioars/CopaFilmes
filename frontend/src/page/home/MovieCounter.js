import React from 'react';
import './MovieCounter.css';

export default function MovieCounter({ count }) {
    return (
        <div className="MovieCounter">
            <span className="MovieCounter__title">Selecionados</span>
            <span className="MovieCounter__counter">{`${count}`} de 8 filmes</span>
        </div>
    );
}

MovieCounter.defaultProps = {
    count: 0
};
