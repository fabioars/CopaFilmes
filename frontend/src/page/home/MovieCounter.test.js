import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import MovieCounter from './MovieCounter';

describe('Movie Counter', () => {
    let container = null;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });


    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <MovieCounter />
            ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with the right number', () => {
        const tree = renderer.create(<MovieCounter count={5}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with string', () => {
        const tree = renderer.create(<MovieCounter count="5"/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
