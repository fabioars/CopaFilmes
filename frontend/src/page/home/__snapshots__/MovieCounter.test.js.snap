// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Counter renders correctly 1`] = `
<div
  className="MovieCounter"
>
  <span
    className="MovieCounter__title"
  >
    Selecionados
  </span>
  <span
    className="MovieCounter__counter"
  >
    0
     de 8 filmes
  </span>
</div>
`;

exports[`Movie Counter renders correctly with string 1`] = `
<div
  className="MovieCounter"
>
  <span
    className="MovieCounter__title"
  >
    Selecionados
  </span>
  <span
    className="MovieCounter__counter"
  >
    5
     de 8 filmes
  </span>
</div>
`;

exports[`Movie Counter renders correctly with the right number 1`] = `
<div
  className="MovieCounter"
>
  <span
    className="MovieCounter__title"
  >
    Selecionados
  </span>
  <span
    className="MovieCounter__counter"
  >
    5
     de 8 filmes
  </span>
</div>
`;
